### ISSUE TITLE
No native build was found - argon2 + node v20.11.0 + node-gyp-build 4.8.1

### ISSUE THREAD

**Issue by gligorot** [prebuild/node-gyp-build#75]
Original issue is from a package that uses `node-gyp-build` - argon2: https://github.com/ranisalt/node-argon2/issues/413

### Actual error (from our full project) 
> No native build was found for platform=darwin arch=arm64 runtime=node abi=115 uv=1 armv=8 libc=glibc node=20.11.0\n    loaded from: xyz/node_modules/argon2\n",

### Error for minimal reproduction testing

 If `build` is missing then `npm install` (quietly!) failed, if present it succeeded. 
```sh
ls node_modules/argon2/build
```

### Minimal reproduction
Examples for npm but yarn has the same behavior.

1. Working (node 20.2.0 always works)
```json
{
  "volta": {
    "node": "20.2.0"
  },
  "dependencies": {
    "argon2": "^0.41.0"
  }
}
```

2. Not working (node 20.3.0, 20.11.0 etc. don't work unless node-gyp-build is fixated)
```json
{
  "volta": {
    "node": "20.11.0"
  },
  "dependencies": {
    "argon2": "^0.41.0"
  }
}
```

3. Working (node 20.3.0, 20.11.0 etc. **AND** node-gyp-build 4.7.1)
```json
{
  "volta": {
    "node": "20.11.0"
  },
  "overrides": {
    "node-gyp-build": "4.7.1"
  },
  "dependencies": {
    "argon2": "^0.41.0"
  }
}
```

### Expected behaviour
`node_modules/argon2/build` is always present, regardless of the specific Node 20 version

### Other details
The resolved version of argon2 in all cases is:
```json
    "node_modules/argon2": {
      "version": "0.41.0",
```
But issue was present on at least`0.40.3` as well.

### Environment
**Operating system**:
 Sonoma 14.6.1 (23G93).
 Apple Silicon (M1 Pro).

**Node version:**
- working: 20.2.0
- not working: 20.3.0, 20.11.0, 20.16.0

**Package manager version:**
- npm: 9.6.7
- yarn: 1.22.18

**Other relevant versions:**
Python: 3.12, installed via pyenv (assuming it matters for node-gyp)

**Comment by gligorot** [comment]
U: node-gyp-build: 4.8.0 doesn't work either, so the issue is somewhere in this 2 line change?
https://github.com/prebuild/node-gyp-build/compare/v4.7.1...v4.8.0

**Comment by gligorot** [comment]
tl;dr issue was with `yarn` and our `yarn.lock` file.

We had two separate node-gyp-build versions requested (as dependency of dependecy):
```sh
❯ yarn why node-gyp-build
├─ @parcel/watcher@npm:2.0.4
│  └─ node-gyp-build@npm:4.5.0 (via npm:^4.3.0)
│
├─ argon2@npm:0.40.3
│  └─ node-gyp-build@npm:4.8.1 (via npm:^4.8.0)
│
└─ leveldown@npm:6.1.1
   └─ node-gyp-build@npm:4.5.0 (via npm:^4.3.0)
``` 

This resolved to 2 different versions in yarn.lock:
```
"node-gyp-build@npm:^4.3.0":
version: 4.5.0
resolution: "node-gyp-build@npm:4.5.0"
...

"node-gyp-build@npm:^4.8.0":
version: 4.8.1
resolution: "node-gyp-build@npm:4.8.1"
...
```

But only one was installed:
```
cat node_modules/node-gyp-build/package.json | grep version // => "version": "4.5.0",
```

### The fix
```
yarn dedupe node-gyp-build
```

Now only one version is correctly resolved and installed:
```
cat node_modules/node-gyp-build/package.json | grep version # => "version": "4.8.1",
```

And we correctly use a prebuilt binary.

**Comment by gligorot** [comment]
tl;dr issue was with `yarn` and our `yarn.lock` file.

We had two separate node-gyp-build versions requested (as dependency of dependecy):
```sh
❯ yarn why node-gyp-build
├─ @parcel/watcher@npm:2.0.4
│  └─ node-gyp-build@npm:4.5.0 (via npm:^4.3.0)
│
├─ argon2@npm:0.40.3
│  └─ node-gyp-build@npm:4.8.1 (via npm:^4.8.0)
│
└─ leveldown@npm:6.1.1
   └─ node-gyp-build@npm:4.5.0 (via npm:^4.3.0)
``` 

This resolved to 2 different versions in yarn.lock:
```
"node-gyp-build@npm:^4.3.0":
version: 4.5.0
resolution: "node-gyp-build@npm:4.5.0"
...

"node-gyp-build@npm:^4.8.0":
version: 4.8.1
resolution: "node-gyp-build@npm:4.8.1"
...
```

But only one was installed:
```
cat node_modules/node-gyp-build/package.json | grep version // => "version": "4.5.0",
```

### The fix
```
yarn dedupe node-gyp-build
```

Now only one version is correctly resolved and installed:
```
cat node_modules/node-gyp-build/package.json | grep version # => "version": "4.8.1",
```

And we correctly use a prebuilt binary.
